ansible/roles/slurm_recompile/defaults/main.yml

```diff
@@ -0,0 +1,3 @@
+---
+slurm_recompile_nvml: false
```

Comment:
There's no docs for this role, which is maybe OK, but I also can't work out what this does? Could you at least add a comment here please?

It looks like it changes whether we recompile slurm against nvml or not, but if that was the case:
a) I'd expect it to be maybe `slurm_recompile_with_nvml` for clarity
b) Surely we either need the default to be `true`, or it needs to be set in inventory or something?

so given that, maybe I've misunderstood?

Reply:
The role will recompile all the slurm packages and install them (replacing the ones from the package repos). The idea is that there may be many reasons why you'd want to recompile: linking against nvml is just one. I've therefore exposed it as a flag and we set it based on whether the builder is a member of the cuda group here: https://github.com/stackhpc/ansible-slurm-appliance/pull/656/files#diff-b54b4dacb4ddd9af2dca5fd7f9b0f5a09f4baa4e430c34e2af554cc7d9548c15R261.

Correct, `slurm_recompile_nvml` essentially sets the --with-nvml argument to configure as part of the compilation process. I think the comment on the naming is a good one.